src/gui/basegraphgl.h

```diff
@@ -28,26 +28,34 @@
 #include <QMouseEvent>
 #include <QMutex>
 #include <QPainter>
+#include <QListWidget>
 #include <QTimer>
 
 #include "core/experiment.h"
 
 #include "colormap.h"
 #include "experimentwidget.h"
-#include "graphwidget.h"
 #include "maingui.h"
-#include <QListWidget>
 
 class Ui_BaseGraphGL;
 
 namespace evoplex {
 
+class GraphWidget;
+
 enum class CacheStatus {
     Ready,
     Updating,
     Scheduled
 };
 
+enum class SelectionMode {
+    Default = -1,   // used in experiment designer, opens the 'light' inspector
+    Select = 0,     // Select mode
+    NodeEdit = 1,   // Node tool
+    EdgeEdit = 2    // Edge tool
+};
```

Comment:
4 modes? what's the difference between `Default` and `Select`?

https://github.com/evoplex/evoplex/pull/43#discussion_r309886233

Reply:
They will probably be merged into one eventually (if/when a "toggle" button for the light inspector is implemented). I created 2 different modes because it allows me to keep the code for the full inspector and the light inspector completely separate making it easier to write and debug

Reply:
what if I open the full inspector in an experiment?

what if I have the full inspector opened while adding nodes? I would expect the inspector to be refreshed to show the info of the current node, wouldn't you expect the same?

so, I think that the "full/light" inspector logic is in another domain and should not be expressed in this enum. Actually, the `Select` mode should just mean that the graph structure is "read-only", i.e., mouse click doesn't change anything. However, it doesn't mean that the inspector will never be filled with the information of the current selection when the `mode != Select`

in this way, I think the enum should have 3 states and the basegraphgl should not be responsible for doing anything with the inspectors directly, i.e., we could just have a signal to let the inspectors know that the current selection changed `emit selectionChanged(id, type)`

Reply:
if you just want to temporarily disable the light inspector, for now, you could add a bool like `mIsFullInspectorOpen`

Reply:
Hm I agree but the one thing that makes me skeptical is multi-selection. Now, multi-selection is a must for the full inspector to work properly, otherwise, it is simply a re-skinned light inspector, but what about multi-selection in "read-only" mode? It doesn't make much sense since we wouldn't be able to edit or move the nodes and simply displaying a list of ids isn't much helpful either. So how should the full inspector behave when selecting multiple nodes? Ignore it, display only the inspector for the last selected node or not allow multi-selection in the experiment designer in the first place?

Reply:
> Now, multi-selection is a must for the full inspector to work properly, otherwise, it is simply a re-skinned light inspector

I don't think so.
The light inspector (should) only show the state of the plotted attribute (instead of everything). Also, I think we don't need to show the ids of the connected edges in the light inspector. 

Thus, I don't think that "multi-selection" is the main reason to implement the full inspector. Actually, the full inspector would be handy even if we don't support multi-selection at all (which is not the case, sure)

> So how should the full inspector behave when selecting multiple nodes?

I think you tackled it on your last mock-up, don't you? The layout for single/multi selection is exactly the same --- the only difference is that you display a `*` if the values are different...

Reply:
Sorry my mistake, I meant to ask how the light inspector is supposed to behave when selecting multiple nodes (should it display all the node ids or simply the last one)

Reply:
I think the behaviour should be the same. However, instead of a (ugly) list, we could just show the ids in plain text...
```
Node ids: 1, 2, 3, 4
AttrName: value (or *)
```

Reply:
@cardinot Sounds good! I removed the 'default' tool